lib/ansible/module_utils/basic.py

```diff
@@ -1993,7 +1993,7 @@ def atomic_move(self, src, dest, unsafe_writes=False):
                             try:
                                 try:
                                     out_dest = open(b_dest, 'wb')
-                                    in_src = open(b_src, 'rb')
+                                    in_src = open(b_tmp_dest_name, 'rb')
```

Comment:
You could either do an os.path.exists check here to decide based on whether b_src exists or you could abstract the unsafe_writes fallback into a helper method, and then write two try: excepts that target each section of code.  If the code fails in  shutil.copy2 pr shutil.move then call the helper function with b_src.  If it fails in os.rename then call the helper with b_tmp_dest_name.  atomic_move is getting rather long so the latter may be the better option.